benchmarking/commons/benchmark_definitions/yahpo.py

```diff
@@ -435,32 +473,1524 @@ def yahpo_rbv2_svm_benchmark(dataset_name, metric, mode="max"):
         blackbox_name="yahpo-rbv2_svm",
         dataset_name=dataset_name,
         max_resource_attr="max_resources",
+        surrogate_kwargs=surrogate_kwargs,
     )
 
 
-yahpo_rbv2_svm_benchmark_definitions = {
-    f"yahpo-rbv2_svm_{metric}-{name}": yahpo_rbv2_svm_benchmark(name, metric, mode)
-    for name in yahpo_rbv2_instances
-    for metric, mode in yahpo_rbv2_metrics
-}
-
-
-def yahpo_rbv2_aknn_benchmark(dataset_name, metric, mode="max"):
+def yahpo_rbv2_aknn_benchmark(
+    dataset_name,
+    metric,
+    mode="max",
+    restrict_fidelities=False,
+    max_wallclock_time=900,
+):
+    surrogate_kwargs = (
+        {
+            "fidelities": yahpo_rbv2_sampled_fidelities,
+        }
+        if restrict_fidelities
+        else None
+    )
     return SurrogateBenchmarkDefinition(
-        max_wallclock_time=1800,  # TODO
-        max_num_evaluations=1000,
+        max_wallclock_time=max_wallclock_time,
         n_workers=4,
         elapsed_time_attr="timetrain",
         metric=metric,
         mode=mode,
         blackbox_name="yahpo-rbv2_aknn",
         dataset_name=dataset_name,
         max_resource_attr="max_resources",
+        surrogate_kwargs=surrogate_kwargs,
     )
 
 
-yahpo_rbv2_aknn_benchmark_definitions = {
-    f"yahpo-rbv2_aknn_{metric}-{name}": yahpo_rbv2_aknn_benchmark(name, metric, mode)
-    for name in yahpo_rbv2_instances
-    for metric, mode in yahpo_rbv2_metrics
+yahpo_rbv2_benchmark = {
+    "rpart": yahpo_rbv2_rpart_benchmark,
+    "glmnet": yahpo_rbv2_glmnet_benchmark,
+    "ranger": yahpo_rbv2_ranger_benchmark,
+    "xgboost": yahpo_rbv2_xgboost_benchmark,
+    "svm": yahpo_rbv2_svm_benchmark,
+    "aknn": yahpo_rbv2_aknn_benchmark,
+}
+
+
+# These are not all instances available in YAHPO rbv2, some filtering has
+# been applied:
+# - f1:
+#   After sampling 100 configs at random, best f1 > 0.2.
+#   Statistics of instances filtered out:
+#   rpart:   max = 0.011020094156265259, avg = 0.0008951445925049484
+#   glmnet:  max = 0.0017474889755249023, avg = 7.839714089641348e-05
+#   ranger:  max = 0.059225380420684814, avg = 0.0039265831001102924
+#   xgboost: max = 0.18943724036216736, avg = 0.015040344558656216
+#   svm:     max = 0.023529857397079468, avg = 0.0020623435266315937
+#   aknn:    max = 0.02313774824142456, avg = 0.0024604059290140867
+# - auc:
+#   After sampling 1000 configs at random, best auc < 0.999
+#   Statistics of instances filtered out:
+#   rpart:   -
+#   glmnet:  min = 0.9990544319152832, avg = 0.999367892742157
+#   ranger:  min = 0.9990753531455994, avg = 0.9997223019599915
+#   xgboost: min = 0.9995461702346802, avg = 0.999969482421875
+#   svm:     min = 0.999025821685791, avg = 0.9996441006660461
+#   aknn:    min = 0.9991077780723572, avg = 0.9997339844703674
+yahpo_rbv2_instances = {
```

Comment:
The data stored from here to L1912 should probably be stored in resource files (csv, json , etc.) rather than in function defintions.

Reply:
I don't think this added complexity buys anything here. These variables are not automatically generated.